### DISCUSSION TITLE
Dictionary Merge aborts with lists

### DISCUSSION THREAD

**Discussion by stevebleazard** [biojppm/rapidyaml#254] [General]
I came across this while testing dictionary merge (<<:) with python
```python
import ryml

src = """
A: &A
    V: 3
    L:
        - 1
B:
    <<: *A
    V: 4
    L:
        -5
"""
tree = ryml.parse_in_arena(src)
tree.resolve()
```
Running this results in 
`/tmp/pip-req-build-_e_q4qdk/src/c4/yml/tree.cpp:969:check_failed: (has_sibling(node, after) && has_sibling(after, node))`
Removing the `L:` from the second (`B:`) entry fixes the problem. I am not sure if this is an issue or not.

**Comment by biojppm** [comment]
* on the `A` map, is `L` supposed to be a list?
* on the `B` map, is `L` supposed to be a scalar?

~~merging a list onto a scalar is currently not possible~~ [EDIT] I think the problem may be a different one. I'll need some time to look at this.

But the question still stands: are `tree["A"]["L"]` and `tree["B"]["L"]` supposed to be different types?

**Reply by stevebleazard** [reply]
B's L should be a list, this appears to be a pasting error:
```
B:
    <<: *A
    V: 4
    L:
        - 5
```

**Comment by biojppm** [comment]
[Looking at the spec](https://yaml.org/type/merge.html) which says that earlier nodes prevail (ie *"each of its key/value pairs is inserted into the current mapping, unless the key already exists in it"*), the `<<: *A` is pointless because the nodes that it brings are all pre-existing in the destination.

Still, even in that case it seems that there is a problem in ryml.

**Reply by stevebleazard** [reply]
You are correct on the `<<: A` being pointless, it's just intended as a minimal example of the behaviour.
